File: GameBox.Data/Models/Game.cs
using GameBox.Core;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GameBox.Data.Models
{
    public class Game
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }

        [Required]
        [MinLength(Constants.GameConstants.TitleMinLength)]
        [MaxLength(Constants.GameConstants.TitleMaxLength)]
        public string Title { get; set; }

        [Range(
            Constants.GameConstants.MinPrice,
            Constants.GameConstants.MaxPrice)]
        public decimal Price { get; set; }

        [Range(
            Constants.GameConstants.MinSize,
            Constants.GameConstants.MaxSize)]
        public double Size { get; set; }

        [Required]
        [MinLength(Constants.GameConstants.MinVideoIdLength)]
        [MaxLength(Constants.GameConstants.MaxVideoIdLength)]
        public string VideoId { get; set; }

        public string ThumbnailUrl { get; set; }

        [Required]
        [MinLength(Constants.GameConstants.MinDescriptionLength)]
        public string Description { get; set; }

        public DateTime ReleaseDate { get; set; }

        public int ViewCount { get; set; }

        public Guid CategoryId { get; set; }

        public Category Category { get; set; }

        public List<GameOrder> Orders { get; set; } = new List<GameOrder>();
    }
}

File: Client/game_box_mobile_ui/lib/models/game/game.dart
class Game {
  String id;
  String title;
  double price;
  double size;
  String videoId;
  String? thumbnailUrl;
  String description;
  int viewCount;

  Game({
    required this.id,
    required this.title,
    required this.price,
    required this.size,
    required this.videoId,
    required this.thumbnailUrl,
    required this.description,
    required this.viewCount,
  });

  factory Game.fromJson(Map<String, dynamic> json) {
    return Game(
      id: json['id'] as String,
      title: json['title'] as String,
      price: json['price'] as double,
      size: json['size'] as double,
      videoId: json['videoId'] as String,
      thumbnailUrl: json['thumbnailUrl'] == null ? '' : json['thumbnailUrl'] as String,
      description: json['description'] as String,
      viewCount: json['viewCount'] as int,
    );
  }
}
